File: node-scraper/produto_scraper.js
const axios = require("axios");
const cheerio = require("cheerio");

class Produto{
    constructor(url){
        this.url = url;
        this.titulo = null;
        this.descricao = null;
        this.preco = null;
        this.imagem = null;
    }
    async extrair_dados() {
        try{
            const response = await axios.get(this.url);
            const html = response.data;
            const $ = cheerio.load(html);
            this.titulo = $('.showcase__description > h1').text().trim();
            this.descricao = $('.feature__main-content').text().trim();
            this.preco = $('.saleInCents-value').first().text().trim();
            this.imagem = $('.image-presenter__carousel img').attr('src');
        }
        catch (erro) {
            console.error("Erro:", erro.message);
        }
    }
}

(async() => {
    const url = 'https://www.netshoes.com.br/p/mochila-nike-heritage-preto+branco-2IC-3604-026';
    const produto = new Produto(url);
    await produto.extrair_dados();  
    console.log(`Título do produto: ${produto.titulo}\n`);  
    console.log(`Descrição: ${produto.descricao}\n`);
    console.log(`Preço: ${produto.preco}\n`);
    console.log(`Url da imagem: ${produto.imagem}`);
})()

File: python-scraper/produto_scraper.py
import requests
from bs4 import BeautifulSoup

def requisicao(url):
    # headers para simular comportamento humano e prevenir o site de barrar a requisição por suspeita de bot
    headers = {
        'User-Agent': (
            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) '
            'AppleWebKit/537.36 (KHTML, like Gecko) '
            'Chrome/115.0.0.0 Safari/537.36'
        ),
        'Accept-Language': 'pt-BR,pt;q=0.9,en-US;q=0.8,en;q=0.7',
        'Referer': 'https://www.google.com/',
        'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8'
        }
    response = requests.get(url, headers=headers)

    if response.status_code == 200:
        dados_pagina = BeautifulSoup(response.text, 'html.parser')
        return dados_pagina
    else:
        print(f'Erro: {response.status_code}')

def extrair_dados(dados):
    titulo = dados.find('h1', class_="product-name").text.strip()
    preco = dados.find('span', class_="saleInCents-value").text.strip()
    descricao = dados.find('div', class_="feature__main-content").text.strip()
    imagem = dados.find('img', alt="Mochila Nike Heritage")['src']
    return imagem

if __name__ == '__main__':
    url = 'https://www.netshoes.com.br/p/mochila-nike-heritage-preto+branco-2IC-3604-026'
    dados_pagina = requisicao(url)
    divs = extrair_dados(dados_pagina)
    # for div in divs:
    print(divs)